storage/src/main/java/tech/pegasys/teku/storage/client/RecentChainData.java

```diff
@@ -242,7 +242,12 @@ private void updateChainHead(
         LOG.info("Skipping head block update to avoid potential rollback of the chain head.");
         return;
       }
+      if (originalHead.isPresent() && originalHead.get().equals(newChainHead)) {
+        LOG.trace("Not updating chain head");
```

Comment:
nit: Probably need to make this message clearer:
```suggestion
        LOG.trace("Skippping head update because new head is same as previous head");
```